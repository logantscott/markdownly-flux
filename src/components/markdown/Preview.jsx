import React from 'react';
import styles from 'github-markdown-css/github-markdown.css';
import { getCurrentFilePreview } from '../../selectors/markdownSelectors';
import { useSelector } from '../../hooks/markdownContext';


const Preview = () => {
  const __html = useSelector(getCurrentFilePreview);

  return (
    <div className={styles['markdown-body']} dangerouslySetInnerHTML={{ __html }}></div>
  );
};

export default Preview;
