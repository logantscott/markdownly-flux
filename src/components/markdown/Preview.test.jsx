import React from 'react';
import { shallow } from 'enzyme';
import Preview from './Preview';
import AppProvider from '../../providers/appProvider';

describe('Preview component', () => {
  it('renders Preview', () => {
    const wrapper = shallow(
      <AppProvider>
        <Preview />
      </AppProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
