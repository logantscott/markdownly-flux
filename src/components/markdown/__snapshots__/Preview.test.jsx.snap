// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview component renders Preview 1`] = `
<ContextProvider
  value={
    Object {
      "dispatch": [Function],
      "state": Object {
        "currentFileID": "ID2938248",
        "files": Object {
          "ID24558248": Object {
            "body": "last test file",
            "title": "one-more",
          },
          "ID2938248": Object {
            "body": "this is a markdown file",
            "title": "test",
          },
          "ID392045": Object {
            "body": "another markdown file",
            "title": "another-test",
          },
        },
      },
    }
  }
>
  <Preview />
</ContextProvider>
`;
